architecture/2025-04-compute-platform-extensibility.md

```diff
@@ -0,0 +1,505 @@
+
+This architecture supports multi-cloud deployments with a unified programming model, where applications can be defined once and deployed to either Kubernetes or Azure Container Instances environments based on the environment configuration.
+
+#### Understanding the Radius/ACI Architecture Relationships
+
+#### Renderer Architecture
+
+The rendering system is a key part of the architecture with several important relationships:
+
+1. Multiplexed Rendering
+    * The `mux.Renderer` acts as a router that directs rendering requests to the appropriate platform-specific renderer
+    * It examines the `EnvironmentComputeKind` (Kubernetes or ACI) from the environment configuration and selects the correct renderer chain
+1. Layered Rendering
+    * Renderers are organized in a layered pattern using composition:
+   KubernetesCompute: kubernetesmetadata → manualscale → daprextension → container
+   ACICompute: aci_manualscale → aci
+    * Each layer handles a specific aspect of rendering, with inner renderers focusing on core functionality and outer renderers adding specialized features
+
+#### Key Type Relationships in the Multiplexer Pattern
+
+1. Core Interface Hierarchy
+    * `renderers.Renderer` is the central interface that defines the rendering contract
+    * All concrete renderers implement this interface, providing a consistent API
+1. Multiplexer Implementation
+    * `mux.Renderer` implements the `renderers.Renderer` interface but delegates to inner renderers
+    * It maintains a map of inner renderers keyed by `EnvironmentComputeKind` (Kubernetes or ACI)
+    * During rendering, it examines the environment compute kind and delegates to the appropriate renderer.
+1. Concrete Renderer Implementations
+    * Platform-specific renderers like `aci.Renderer` and `container.Renderer` (Kubernetes) implement the same interface
+    * Resource-specific renderers like `gateway.Renderer` and `aci_gateway.Renderer` handle specific resource types
+1. Environment Configuration
+    * `EnvironmentCompute` defines the compute environment with its `Kind` property
+    * The multiplexer uses this to determine which inner renderer to invoke
+1. Input and Output Types
+    * `renderers.RenderOptions` provides the context for rendering, including environment configuration
+    * `renderers.RendererOutput` represents the standardized output of the rendering process
+
+This pattern enables the system to:
+
+1. Handle multiple compute environments through a single interface
+1. Dynamically select the appropriate renderer at runtime
+1. Share common interfaces across different platform implementations
+1. Easily extend the system with new compute environments by adding renderers to the map
+
+The design follows the Strategy Pattern, where the rendering strategy is selected based on the environment configuration, while maintaining a consistent interface for the rest of the system.
+
+### Options Considered
+
+#### Proposed Option: Two Milestones
```

Comment:
I think this option makes most sense. The demo already has areas where extensibility would be fairly simple to implement vs the second option proposed